src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -342,4 +395,13 @@ private static boolean violationWeakenedOrEquivalent(LimitViolation violation1,
     List<LfContingency> createContingencies(List<PropagatedContingency> propagatedContingencies, LfNetwork network) {
         return LfContingency.createContingencies(propagatedContingencies, network, network.createDecrementalConnectivity(connectivityProvider), true);
     }
+
+    private void addMonitorInfos(LfNetwork lfNetwork, StateMonitor monitor, Consumer<BranchResult> branchResultConsumer,
+                                 Consumer<BusResults> busResultsConsumer, Consumer<ThreeWindingsTransformerResult> threeWindingsTransformerResultConsumer) {
+        monitor.getBranchIds().forEach(branchId -> branchResultConsumer.accept(lfNetwork.getBranchById(branchId).createBranchResult()));
```

Comment:
I wonder what we should do in case of an invalid branch ID (one that does not existe in the network.

I guess that with this code, it will throw a null pointer exception. I would be in favor of silently ignore that branch instead.
In any case we must add a unit test to cover that case! (same for buses and T3)